pkg/provisioner/resourcehandlers.go

```diff
@@ -279,4 +267,15 @@ func updateObjectBucketPhase(c versioned.Interface, ob *v1alpha1.ObjectBucket, p
 		return (err == nil), err
 	})
 	return
-}
\ No newline at end of file
+}
+
+func validateObjectBucket(ob *v1alpha1.ObjectBucket) error {
+	var errs []string
+	if ob.Spec.Endpoint == nil {
+		errs = append(errs, "Spec.Endpoint cannot be nil")
+	}
+	if ob.Spec.Authentication == nil {
+		errs = append(errs, "Spec.Authentication cannot be nil")
+	}
+	return fmt.Errorf("%s", strings.Join(errs, "; "))
```

Comment:
Seems like this function returns a non nil error even if errs array is empty...
Did you mean this:
```suggestion
	if len(errs) > 0 {
		return fmt.Errorf("%s", strings.Join(errs, "; "))
	}
	return nil
```

Reply:
Right ... wonder why this hasn't triggered in testing.